docs/js/index.js

```diff
@@ -0,0 +1,4 @@
+const settings = JSON.parse(localStorage.getItem(__prefix('__palette')));
```

Comment:
In this script, which runs on every page load, we are reading and applying the scheme preference stored in localStorage by mkdocs-material.

So the flow is:
1. on the first site load the mkdocs-material will set this preference according to the system scheme.
2. on each subsequent page loads it will use the stored setting (hence - didn't reflect the OS-wide scheme switching)
3. user at any time can change the site theme by switching the theme switcher (this also updates localStorage)

And this flow is mainly needed for cases when you use a dark scheme in OS but want to have a light scheme for the site and vice versa.